chains/ethereum/methods.go

```diff
@@ -19,14 +19,15 @@ var gasPrice = big.NewInt(20000000000)
 func (w *Writer) depositAsset(m msg.Message) bool {
 	log15.Info("Handling DepositAsset message", "to", w.conn.cfg.receiver)
 
-	opts, err := w.conn.newTransactOpts(big.NewInt(0), gasLimit, gasPrice)
+	opts, nonce, err := w.conn.newTransactOpts(big.NewInt(0), gasLimit, gasPrice)
 	if err != nil {
 		log15.Error("Failed to build transaction opts", "err", err)
 		return false
```

Comment:
Returning here will leave the lock locked, best to use a `defer` to ensure it's unlocked